Islandora/islandora
# Solr filter for entity relationship that has term.

**GitHub Issue**: n/a

**Other Relevant Links** 
Discussion in yesterday's Open Meeting with Simon: How can we make a Facet that lets you filter by Collection? Right now, all sorts of other compound/Book objects are coming up because they use the same Member Of field.

# What does this Pull Request do?

Makes new Fields available to Search API that let you filter entity reference fields, and only index the target if the target has a given taxonomy term.

# What's new?

* New Search API Processor (needs to be enabled on the Processors page)
* New Fields (mirror existing fields, but are "filtered by term")

* Does this change add any new dependencies?  no
* Does this change require any other modifications to be made to the repository
 (i.e. Regeneration activity, etc.)?  no
* Could this change impact execution of existing code? no
 
# How should this be tested?

* add some content using the member_of field. Have some be member of Collections, and others member of Paged Content. 
* install this PR and clear cache.
* Go to `/admin/config/search/search-api/index/default_solr_index/processors` and enable "Entity Reference, where the target has a given taxonomy term". Save the page.
* Under "Fields", see a new field in the Content section: "Member of (by term URI) [islandora_object] `(entity_reference_with_term__node__islandora_object__field_member_of)`"
* Select it, configure it to only show (books, collections, whatever). Make sure if you're using Collection that you select the same term you applied to your content. You can select multiple terms and say if you want all or any.
* Reindex your content
* Go to Facets, and create a facet for this field you added. (give it a useful name)
* Go to Blocks, and place the block you just created. (find it with the name you gave it above)
* Do a search, and see only the desired "types" in your facet.

# Documentation Status

* Does this change existing behaviour that's currently documented? no
* Does this change require new pages or sections of documentation? maybe.
* Who does this need to be documented for? site builder
* Associated documentation pull request(s): ___  or documentation issue ___

# Additional Notes:



# Interested parties
Tag (@ mention) interested parties or, if unsure, @Islandora/committers
